Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -318,6 +318,35 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
     required public init(for route: Route, routeIndex: Int, routeOptions: RouteOptions, navigationOptions: NavigationOptions? = nil) {
         super.init(nibName: nil, bundle: nil)
         
+        initialize(for: route, routeIndex: routeIndex, routeOptions: routeOptions, navigationOptions: navigationOptions)
```

Comment:
I expected the fix to be in `required public init?(coder aDecoder: NSCoder)`, does it work now?

Reply:
It'll not be possible to provide needed information in `required public init?(coder aDecoder: NSCoder)`. E.g. `route` object should already be available before calling that initializer, which doesn't seem to be possible. Seems that the only way is to initialize all needed objects afterwards and only after that present `NavigationViewController`.

Reply:
Example in this [guide](https://github.com/mapbox/ios-sdk/pull/1155/files#diff-15eb416535da909019f8d0f1dd8faaf5dcaf60dfe33e1e44df91bed2965dc113) shows how new usage can look like.

Reply:
I see your point now, the problem is that if a user forgets to call `initialize` method the app will crash. 😿
But I guess it's the best we can do, hope this problem will be noticeable during the development process.

Reply:
> E.g. `route` object should already be available before calling that initializer, which doesn't seem to be possible.

The separate `initialize` method is really unfortunate. Its name points to a larger problem: most of the stuff that has been happening in the initializer only needs to take place by the time the view controller is presented, right? Ideally we’d move it all into `viewDidLoad()`. As it is, developers using NavigationViewController programmatically have to be careful about prematurely initializing the object. Developers don’t expect initializing a view controller or view to have side effects. I guess the tradeoff is that delaying these steps until `viewDidLoad()` would force us to think about situations where `navigationService` etc. are `nil`, so we’d have to revisit a bunch of implicitly unwrapped optionals.

I guess I’m OK with this approach as a stopgap fix, but in that case we should think about a broader lifecycle cleanup as tail work, with an eye towards deprecating the new `initialize` method.

Reply:
I agree that additional initializer is not an optimal solution. I've changed implementation to use optional parameters for `route`, `routeOptions` `routeIndex` etc. User will be able to change it before `NavigationViewController` presentation. Example can be found [here](https://github.com/mapbox/mapbox-navigation-ios-examples/pull/113).